Tribler/Core/Modules/restapi/downloads_endpoint.py

```diff
@@ -102,3 +119,135 @@ def render_GET(self, request):
                              "max_download_speed": download.get_max_speed(DOWNLOAD)}
             downloads_json.append(download_json)
         return json.dumps({"downloads": downloads_json})
+
+
+class DownloadSpecificEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for dispatching requests to perform operations in a specific discovered channel.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+
+        child_handler_dict = {"remove": DownloadRemoveEndpoint, "stop": DownloadStopEndpoint,
+                              "resume": DownloadResumeEndpoint}
+        for path, child_cls in child_handler_dict.iteritems():
+            self.putChild(path, child_cls(session, bytes(infohash.decode('hex'))))
+
+
+class DownloadRemoveEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for removing downloads from Tribler.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+        self.infohash = infohash
+
+    def render_DELETE(self, request):
+        """
+        .. http:delete:: /download/(string: infohash)/remove
```

Comment:
It would make more sense if the delete was against `/download/(string: infohash)` itself.